File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        sum_value=n
        while True:
            numb=str(sum_value)
            sum_value=0
            for digit in numb:
                sum_value+=int(digit)**2
            if int(numb)==1:
                return True
            elif int(numb)==4:
                return False
            print(numb,sum_value)
        #if sum_value%10==0:
        #    return True
        #return False

File: 202-happy-number/happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    let seen = new Set()
    seen.add(n)
    n = String(n)
    
    while (true){
        let total = 0

        for (digit of n){
            total += Number(digit)**2
        }
        if (total==1){
            return true
        }
        if (seen.has(total)){
            return false
        }
        seen.add(total)
        
        n = String(total)
    }
};